File: src/main/java/com/github/zametki/db/dbi/GroupsDbi.java
package com.github.zametki.db.dbi;

import com.github.zametki.model.Group;
import com.github.zametki.model.GroupId;
import com.github.zametki.model.UserId;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface GroupsDbi {

    void create(@NotNull Group group);

    @NotNull
    List<GroupId> getByUser(@Nullable UserId userId);

    @Nullable
    Group getById(@Nullable GroupId id);

    void update(@NotNull Group c);

    @Nullable
    Group getByName(@NotNull UserId userId, @NotNull String name);
}

File: src/main/java/com/github/zametki/db/dbi/GroupsDbi.kt
package com.github.zametki.db.dbi

import com.github.zametki.model.Group
import com.github.zametki.model.GroupId
import com.github.zametki.model.UserId

interface GroupsDbi {

    fun create(g: Group)

    fun getByUser(userId: UserId?): List<GroupId>

    fun getById(id: GroupId?): Group?

    fun update(g: Group)

    fun getByName(userId: UserId, name: String): Group?

    fun removeEmptyGroup(group: Group)

    fun getSubgroups(id: GroupId?): List<GroupId>
}
